### PR TITLE
[Optimizer] Disable optimizer pass by default

### PR DESCRIPTION
**Pull request by vcanicTT**
Optimizer pass was enabled by default and some tests depend on it. We want to make it disabled by default, and make rest of the compiler independent of the optimizer. Independent means all components should work with and without optimizer, correctness of the output of the compiler doesn't depend on the optimizer.

Issue: https://github.com/tenstorrent/tt-mlir/issues/727

### CODE REVIEW

**Review by odjuricicTT — APPROVED** [review]

File: test/ttmlir/Dialect/TTNN/multiple_add_with_loc.mlir [hunk 1]
```diff
@@ -3,7 +3,7 @@
 #loc = loc("test_ops.py:17_0_0":0:0)
```
Comment (nobradovictt) [inline test/ttmlir/Dialect/TTNN/multiple_add_with_loc.mlir:3]: I added this test, make optimizer enabled here.

File: test/ttmlir/Dialect/TTNN/ttir_to_ttnn_pipeline.mlir [hunk 2]
```diff
@@ -2,7 +2,7 @@
 #any_device = #tt.operand_constraint<dram|l1|scalar|tile|any_device|any_device_tile>
```
Comment (nobradovictt) [inline test/ttmlir/Dialect/TTNN/ttir_to_ttnn_pipeline.mlir:2]: Enable here as well.

**Review by nobradovictt — APPROVED** [review]

File: test/ttmlir/Dialect/TTNN/multiple_add_with_loc.mlir (hunk 1 above)
Comment (vcanicTT) [reply test/ttmlir/Dialect/TTNN/multiple_add_with_loc.mlir:3]: Done.

File: test/ttmlir/Dialect/TTNN/ttir_to_ttnn_pipeline.mlir (hunk 2 above)
Comment (vcanicTT) [reply test/ttmlir/Dialect/TTNN/ttir_to_ttnn_pipeline.mlir:2]: Done.

**Pull request merged**